<a href="https://colab.research.google.com/github/ice4869/Prediction-of-quality-of-Wine/blob/main/%E7%B4%85%E9%85%92%E5%93%81%E8%B3%AA%E5%88%86%E9%A1%9E%E9%A0%90%E6%B8%AC%EF%BC%881%EF%BC%89.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
df = pd.read_csv('https://raw.githubusercontent.com/aniruddhachoudhury/Red-Wine-Quality/master/winequality-red.csv')

fixed acidity（固定酸度）：酒中的非揮發性酸的總含量。

volatile acidity（揮發酸度）：酒中的揮發性酸的總含量。

citric acid（檸檬酸）：酒中的檸檬酸含量。檸檬酸可為酒帶來柑橘風味。

residual sugar（剩餘糖份）：發酵後剩餘的糖份量，以克/升（g/L）為單位。

chlorides（氯化物）：酒中的鹽類含量，以克/升（g/L）為單位。

free sulfur dioxide（游離二氧化硫）：在酒中溶解的游離二氧化硫（SO2）的量，以游離硫酸鹽形式存在。

total sulfur dioxide（總二氧化硫）：游離二氧化硫（SO2）和與其他物質結合的二氧化硫（例如，與酒中的酒石酸結合）總量。

density（密度）：酒的密度，以克/立方厘米（g/cm³）為單位。

pH：酒的酸鹼度。較低的值表示酸性較高，較高的值表示酸性較低。

sulphates（硫酸鹽）：酒中的硫酸鹽含量，以克/升（g/L）為單位。硫酸鹽可用於酒的防腐劑和抗氧化劑。

alcohol（酒精）：酒中的酒精濃度，以百分比體積（% vol）為單位。

quality（品質）：酒的品質評分，通常以整數形式表示（例如，從0到10的等級）。


In [2]:
# 顯示前幾筆資料
print(df.head())

# 觀察每個特徵欄位的資料分佈情況
print(df.describe())

   fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
0            7.4              0.70         0.00             1.9      0.076   
1            7.8              0.88         0.00             2.6      0.098   
2            7.8              0.76         0.04             2.3      0.092   
3           11.2              0.28         0.56             1.9      0.075   
4            7.4              0.70         0.00             1.9      0.076   

   free sulfur dioxide  total sulfur dioxide  density    pH  sulphates  \
0                 11.0                  34.0   0.9978  3.51       0.56   
1                 25.0                  67.0   0.9968  3.20       0.68   
2                 15.0                  54.0   0.9970  3.26       0.65   
3                 17.0                  60.0   0.9980  3.16       0.58   
4                 11.0                  34.0   0.9978  3.51       0.56   

   alcohol  quality  
0      9.4        5  
1      9.8        5  
2      9.8        5 

In [3]:
# 觀察 "quality" 標籤欄位的資料分佈情況
quality_distribution = df['quality'].value_counts().sort_index()
print(quality_distribution)

3     10
4     53
5    681
6    638
7    199
8     18
Name: quality, dtype: int64


從結果中可以觀察到以下情況：

"quality" 的取值範圍從 3 到 8。
大部分的紅酒樣本分布在品質評分 5 和 6 的範圍內，分別有 681 和 638 筆樣本。

品質評分為 3、4、7 和 8 的樣本數量相對較少，分別為 10、53、199 和 18 筆樣本。